File: front/app/types/voo.ts
export class Voo{
    constructor(
        public id:number | null,
        public codigoVoo:string,
        public capacidade:number,
        public assentosDisponiveis:number,
        public cidadeOrigem:string,
        public cidadeDestino:string,
        public dataHoraSaida:string,
        public dataHoraChegada:string,
        public status:string
    ){}
}

export interface VooFormProps{
    vooExistente?:Voo
}

File: src/main/java/com/example/aeropass/entities/Voo.java
package com.example.aeropass.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Voo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    // public Long idAviao;

    public String codigoVoo;

    public Long capacidade;

    public Long assentosDisponiveis;

    public String cidadeOrigem;

    public String cidadeDestino;

    public LocalDateTime dataHoraSaida;

    public LocalDateTime dataHoraChegada;

    private EnumStatusVoo status = EnumStatusVoo.ATIVO;
}
